### PR TITLE
Add import from a timefreq.mat from fieldtrip

### PR DESCRIPTION
**Pull request by AlexChalard**
Adding a callback to import & convert a timefreq.mat file obtain from fieldtrip.
This is a first attempt, if is ok, I will add some additional options and create a channel.mat to allow the topographic display.

### CODE REVIEW

File: toolbox/io/import_timefreq_ft.m [hunk 1]
```diff
@@ -0,0 +1,41 @@
+function OutputFile = import_timefreq_ft(iStudy)
```
Comment (ftadel) [inline toolbox/io/import_timefreq_ft.m]: To respect the conventions in the existing code: functions import_*.m are generic for a type of data, and should include a switch between various file formats. Simple examples include import_video.m or import_dipoles.m.
Therefore this function should be renamed import_timefreq.m and have the same structure as these example functions.

Comment (ftadel) [reply toolbox/io/import_timefreq_ft.m]: This function could also include the creation of a channel file when there is no channel file in the destination study.

File: toolbox/io/in_timefreq_fieldtrip.m [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+%
+% For more information type "brainstorm license" at command prompt.
+% =============================================================================@
+%
+% Authors: Alexandre Chalard
+
+% Get format
+[fPath, fBase, fExt] = bst_fileparts(DataFile);
+% Initialize returned structure
+DataMat = db_template('timefreqmat');
```
Comment (ftadel) [inline toolbox/io/in_timefreq_fieldtrip.m]: Replace variable name DataMat with TfMat or TimefreqMat, to follow the naming conventions in other functions.
DataMat is mostly used to store MEG/EEG recordings (single trials or averaged).

File: toolbox/io/in_timefreq_fieldtrip.m [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+nChannels = length(ftMat.label);
+DataMat.ChannelFlag = ones(nChannels, 1);
+
+% Mandatory field for bst 
+DataMat.TF        = permute(ftMat.powspctrm,[1 3 2]);
+DataMat.TFmask    = ones(size(DataMat.TF,3),size(DataMat.TF,2));
+DataMat.Time      = ftMat.time;
+DataMat.Freqs     = ftMat.freq;
+DataMat.RowNames  = ftMat.label;
+DataMat.Method    = 'From FieldTrip';
```
Comment (ftadel) [inline toolbox/io/in_timefreq_fieldtrip.m]: Can't you get what method was used to compute this file from the provenance cfg structure? That would be more informative.
Avoid spaces and capital letters in this field.

Comment (ftadel) [reply toolbox/io/in_timefreq_fieldtrip.m]: The field Measure should be provided as well ('log','magnitude','power','phase','none')

File: toolbox/io/in_timefreq_fieldtrip.m [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+%
+% Authors: Alexandre Chalard
+
+% Get format
+[fPath, fBase, fExt] = bst_fileparts(DataFile);
+% Initialize returned structure
+DataMat = db_template('timefreqmat');
+DataMat.Comment  = fBase;
+DataMat.Device   = 'FieldTrip';
+DataMat.DataType = 'timefreq';
```
Comment (ftadel) [inline toolbox/io/in_timefreq_fieldtrip.m]: DataType represents the type of data the file was computed from: 'data' (for recordings), 'results' (sources), 'scout' or 'matrix'.
See documentation here: 
https://neuroimage.usc.edu/brainstorm/Tutorials/TimeFrequency#On_the_hard_drive

File: toolbox/io/in_timefreq_fieldtrip.m [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+% =============================================================================@
+%
+% Authors: Alexandre Chalard
+
+% Get format
+[fPath, fBase, fExt] = bst_fileparts(DataFile);
+% Initialize returned structure
+DataMat = db_template('timefreqmat');
+DataMat.Comment  = fBase;
+DataMat.Device   = 'FieldTrip';
```
Comment (ftadel) [inline toolbox/io/in_timefreq_fieldtrip.m]: There is no Device field in the timefreq structure.

File: toolbox/io/in_timefreq_fieldtrip.m (hunk 3 above)
Comment (ftadel) [reply toolbox/io/in_timefreq_fieldtrip.m]: In general, the more fields you can fill, the better.

File: toolbox/io/in_timefreq_fieldtrip.m (hunk 4 above)
Comment (ftadel) [reply toolbox/io/in_timefreq_fieldtrip.m]: Probably here should be added a switch to handle the different types of time-frequency files that can be imported from FieldTrip: sensor space vs. source space vs. ROI.

**Comment by AlexChalard** [comment]
Dear Francois,

Thank you to take time to review the code.

How should I proceed with github to take into account your review and proceed the change in the code ? 

Kind regards,

Alex

**Comment by ftadel** [comment]
I think if you simply push your changes to your fork, the modifications should appear automatically in this pull request.

**Comment by ftadel** [comment]
@AlexChalard  Do you still want to merge this PR?

**Comment by AlexChalard** [comment]
Hello @ftadel,
Currently I'm not working on it, you can close this PR.

Best, 

Alex

**Pull request closed without merging**